Assets/MixedRealityToolkit/Utilities/Editor/Setup/MixedRealityProjectConfigurator.cs

```diff
@@ -0,0 +1,249 @@
+        }
+
+        /// <summary>
+        /// Checks if current Unity project has latest scripting runtime
+        /// </summary>
+        public static bool IsLatestScriptingRuntime()
+        {
+#if !UNITY_2019_3_OR_NEWER
+            return PlayerSettings.scriptingRuntimeVersion == ScriptingRuntimeVersion.Latest;
+#else
+        return true;
+#endif
+        }
+
+        /// <summary>
+        /// Configures current Unity project to use latest scripting runtime and reloads project
+        /// </summary>
+        public static void SetLatestScriptingRuntime()
+        {
+#if !UNITY_2019_3_OR_NEWER
+            PlayerSettings.scriptingRuntimeVersion = ScriptingRuntimeVersion.Latest;
+            EditorApplication.OpenProject(Directory.GetParent(Application.dataPath).ToString());
+#endif
+        }
+
+        /// <summary>
+        /// Checks if current Unity projects uses force text serialziation
+        /// </summary>
+        public static bool IsForceTextSerialization()
+        {
+            return EditorSettings.serializationMode == SerializationMode.ForceText;
+        }
+
+        /// <summary>
+        /// Configures current Unity project to force text serialization
+        /// </summary>
+        public static void SetForceTextSerialization()
+        {
+            EditorSettings.serializationMode = SerializationMode.ForceText;
+        }
+
+        /// <summary>
+        /// Checks if current Unity project uses visible meta files
+        /// </summary>
+        public static bool IsVisibleMetaFiles()
+        {
+            return EditorSettings.externalVersionControl.Equals("Visible Meta Files");
+        }
+
+        /// <summary>
+        /// Configures current Unity project to enabled visible meta files
+        /// </summary>
+        public static void SetVisibleMetaFiles()
+        {
+            EditorSettings.externalVersionControl = "Visible Meta Files";
+        }
+
+        /// <summary>
+        /// Checks if current Unity project has the default Spatial Awareness layer set in the Layers settings
+        /// </summary>
+        public static bool HasSpatialAwarenessLayer()
+        {
+            return !string.IsNullOrEmpty(LayerMask.LayerToName(SpatialAwarenessDefaultLayer));
+        }
+
+        /// <summary>
+        /// Configures current Unity project to contain the default Spatial Awareness layer set in the Layers settings
+        /// </summary>
+        public static void SetSpatialAwarenessLayer()
+        {
+            if (!HasSpatialAwarenessLayer())
+            {
+                if (EditorLayerExtensions.SetupLayer(SpatialAwarenessDefaultLayer, "Spatial Awareness"))
+                {
+                    Debug.LogWarning(string.Format($"Can't modify project layers. It's possible the format of the layers and tags data has changed in this version of Unity. Set layer {SpatialAwarenessDefaultLayer} to \"Spatial Awareness\" manually via Project Settings > Tags and Layers window."));
```

Comment:
This message is misleading if the layer is already set to Spatial Awareness.  Can we get a wording tweak or have it succeed if the value is already set?

Another option would be to differentiate "failure to set", "already set" and "set to something we didnt expect"